src/contest/weglide/UploadIGCFile.cpp

```diff
@@ -0,0 +1,207 @@
+ * @param key   - the key of the wanted value as string
+ * @param error_code - the return value in case of faulty json object
+ *
+ * @return the converted string value or the error_code (in case of an error)
+ */
+
+static const UTF8ToWideConverter 
+GetJsonString(boost::json::standalone::value json_value, std::string_view key,
+              const char* error_code = "") noexcept {
+  auto value = json_value.at(key);
+  const auto value_str = (value.is_string() ?
+    value.get_string().c_str() : error_code);
+  // const UTF8ToWideConverter value_str2(value_str);
+  return UTF8ToWideConverter(value_str); //  UTF8ToWideConverter(value_str.str());
+}
+
+namespace WeGlide {
+
+struct User {
+  uint32_t id = 0;
+  BrokenDate birthdate;
+  StaticString<0x80> name;
+};
+
+struct Aircraft {
+  uint32_t id = 0;
+  StaticString<0x40> name;
+  StaticString<4> kind;  // 'MG' - motor aircraft,...
+  StaticString<10> sc_class;
+};
+
+struct UploadResponse {
+  uint64_t flight_id = 0;
+  User user;
+  Aircraft aircraft;
+  StaticString<0x40> scoring_date;
+  StaticString<0x40> registration;
+  StaticString<0x40> competition_id;
+};
+
+static void
+UploadJsonInterpreter(const boost::json::value &json,
+                      UploadResponse& response) {
+  if (json.is_array() && (json.get_array().size() > 0)) {
+    auto flight_obj = json.get_array().at(0);
+    response.scoring_date = GetJsonString(flight_obj, "scoring_date").c_str();
+    response.flight_id = flight_obj.at("id").to_number<int64_t>();
+    response.registration = GetJsonString(flight_obj, "registration").c_str();
+    response.competition_id = GetJsonString(flight_obj, "competition_id").c_str();
+    auto user = flight_obj.at("user");
+    if (user.is_object()) {
+      auto user_obj = user.as_object();
+      response.user.id = user_obj.at("id").to_number<uint32_t>();
+      response.user.name = GetJsonString(user_obj, "name").c_str();
+    }
+    auto aircraft = flight_obj.at("aircraft");
+    if (aircraft.is_object()) {
+      auto aircraft_obj = aircraft.as_object();
+      response.aircraft.id = aircraft_obj.at("id").to_number<uint32_t>();
+      response.aircraft.name = GetJsonString(aircraft_obj, "name").c_str();
+      response.aircraft.kind = GetJsonString(aircraft_obj, "kind").c_str();
+      response.aircraft.sc_class = GetJsonString(aircraft_obj, "sc_class").c_str();
+    }
+  }
+  else {
+    throw std::runtime_error("The http response has a wrong JSON format!");
+  }
+}
+
+static void
+UploadSuccessDialog(const UploadResponse& response, const Path igc_path) {
+  // TODO: Create a real Dialog with fields
+  // With this Dialog insert the possibilty to update/patch the flight
+  // f.e. copilot in double seater, scoring class, short comment and so on
+  StaticString<0x1000> msg;
+  msg.Format(_("File upload '%s' was succesfull"),
+    igc_path.c_str());
+  msg.append(_T("!\n\n"));
+  msg.AppendFormat(_T("%-20s: %u\n"), _("flight_id"), response.flight_id);
```

Comment:
You're doing layout by inserting spaces - this is very ugly, and doesn't work with proportional fonts.
How should a translator translate "flight_id"?
Why are you doing many `AppendFormat()` calls instead of just one?
This whole piece of code is bad.

Reply:
Layout w/o spaces
'flight_id' is 'Flight ID' - done with all the others!
I use only one `msg.Format` (I thought with many `AppenFormat()` we have it would be better structured - but of course compile time and run time...)
> This whole piece of code is bad.

This should only be a first 'shot', in the next step I would like to include a better dialog (with the option to also change some values ​​on the server)

Reply:
Just ... omit the bad code. Having this huge multi-line dialog message with flight_id etc. seems rather useless. So I wonder why you took all the time to write it, in a way that's bad. Just concentrate on the important pieces, and omit stuff you can do later. I keep wondering why you make it so hard for yourself.

Reply:
You may be right about that, but I wanted at least some decent feedback on how the flight went down on WeGlide, at least until there's something better!